Assets/Scripts/Characters/EnemyController.cs

```diff
@@ -15,15 +17,47 @@ public class EnemyController : SerializedMonoBehaviour
         [OdinSerialize, HideLabel]
         public EnemyState State { get; set; }
 
+        public Canvas enemyStateCanvas;
+        public GameObject enemyStateIndicator;
+        public Sprite enemyStatePatrol;
+        public Sprite enemyStateInvestigate;
+        public Sprite enemyStateChase;
+
+        private Camera _camera;
+        
         private void Awake()
         {
             State = gameObject.GetComponent<Patrol>();
+            enemyStateIndicator.GetComponent<Image>().sprite = enemyStatePatrol;
+        }
+
+        private void Start()
+        {
+            _camera = Camera.main;
+        }
+
+        private void Update()
+        {
+            enemyStateCanvas.transform.rotation = _camera.transform.rotation;
         }
 
         public async void FixedUpdate()
         {
             await State.TickState();
             State.UpdateState();
+
+            if (State == gameObject.GetComponent<Patrol>())
+            {
+                enemyStateIndicator.GetComponent<Image>().sprite = enemyStatePatrol;
+            }
+            else if (State == gameObject.GetComponent<Investigate>())
+            {
+                enemyStateIndicator.GetComponent<Image>().sprite = enemyStateInvestigate;
+            }
+            else if (State == gameObject.GetComponent<Chase>())
+            {
+                enemyStateIndicator.GetComponent<Image>().sprite = enemyStateChase;
+            }
```

Comment:
This code can be made more readable using `switch` expressions.

```suggestion
            enemyStateIndicator.GetComponent<Image>().sprite = State switch
            {
                Patrol => enemyStatePatrol,
                Investigate => enemyStateInvestigate,
                Chase => enemyStateChase,
                _ => null
            };
```

Reply:
This has been added.